File: apollo/src/main/java/io/muun/apollo/data/os/Configuration.java
package io.muun.apollo.data.os;

import android.content.Context;
import android.content.res.AssetManager;
import android.content.res.Resources;

import java.io.IOException;
import java.util.Properties;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class Configuration {

    private static final String CONFIG_FILE_NAME = "config.properties";

    private final Properties properties;

    /**
     * Constructor.
     */
    @Inject
    public Configuration(Context context) {

        final Resources resources = context.getResources();
        final AssetManager assetManager = resources.getAssets();

        properties = new Properties();

        try {
            properties.load(assetManager.open(CONFIG_FILE_NAME));
        } catch (IOException e) {
            throw new RuntimeException("Config file '" + CONFIG_FILE_NAME + "' not found.", e);
        }
    }

    public long getLong(String key) {

        return Long.parseLong(properties.getProperty(key));
    }

    public int getInt(String key) {

        return Integer.parseInt(properties.getProperty(key));
    }

    public boolean getBoolean(String key) {

        return Boolean.parseBoolean(properties.getProperty(key));
    }

    public String getString(String key) {

        return properties.getProperty(key);
    }
}

File: android/apolloui/src/main/java/io/muun/apollo/data/os/Configuration.kt
package io.muun.apollo.data.os

import android.content.Context
import java.io.IOException
import java.lang.Boolean.*
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class Configuration @Inject constructor(context: Context) {

    companion object {
        private const val CONFIG_FILE_NAME = "config.properties"
    }

    private val properties: Properties = Properties()

    /**
     * Constructor.
     */
    init {
        try {
            properties.load(context.resources.assets.open(CONFIG_FILE_NAME))
        } catch (e: IOException) {
            throw RuntimeException("Config file '$CONFIG_FILE_NAME' not found.", e)
        }
    }

    fun getLong(key: String): Long =
        properties.getProperty(key).toLong()

    fun getInt(key: String): Int =
        properties.getProperty(key).toInt()

    fun getBoolean(key: String): Boolean =
        parseBoolean(properties.getProperty(key))

    fun getString(key: String): String =
        properties.getProperty(key)
}